samples/boards/96b_argonkey/microphone/src/OpenPDMFilter.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * @file    OpenPDMFilter.c
```

Comment:
From my understanding, this should be made available as a lib (such as ext/hal/st/lib/sensor/vl53l0x/) and accessed through a specific driver (cf: drivers/sensor/vl53l0x), then used from this sample.
This should be done in several commits and the commit for lib should provide more info (can't find the link in zephyr doc, here is an example):
```
    title
    
   body    

    Origin: ST Microelectronics
    License: ..
    URL: where to get it
    Commit: (or version)
    Purpose: 
    Maintained-by: External
    
    Signed-off-by: ....
```

Reply:
OK, I agree to move the OpenPDMFilter library under ext/hal/st/lib.

I'm wondering if the microphone driver is really needed. I was more thinking on
using the lib API in the application itself. A driver is maybe too much, as it is
just a passive element that starts sending a PCM stream as soon as it is powered and clocked.
It does not have registers to configure it.

Reply:
Since this is not Apache 2.0 licensed code, check out https://docs.zephyrproject.org/latest/contribute/contribute_guidelines.html#contributing-non-apache-2-0-licensed-components
@nashif @galak

Reply:
Thanks @dbkinder 
I had already discussed this point with @erwango in pvt, but this link is pretty useful!

Reply:
@dbkinder  @erwango 
I had modified the OpenPDMFilter library in order to be compliant with zephyr coding rules, but now I have this opinion to revert it back to the original version from Mbed os. I guess this is the right way to do it.